cmd/queue/main.go

```diff
@@ -104,22 +106,27 @@ func connectStatSink() {
 	autoscalerEndpoint := fmt.Sprintf("ws://%s.%s.svc.cluster.local:%s",
 		servingAutoscaler, system.Namespace, servingAutoscalerPort)
 	logger.Infof("Connecting to autoscaler at %s.", autoscalerEndpoint)
-	for {
-		// TODO: use exponential backoff here
-		time.Sleep(time.Second)
 
+	wait.ExponentialBackoff(wait.Backoff{
+		Duration: 100 * time.Millisecond,
+		Factor:   1.3,
+		Steps:    math.MaxInt32,
```

Comment:
MaxInt could result in some pretty crazy backoff timeouts. I wonder if it might make more sense to backoff for some number of tries, then return an error. The caller can then decide what to do:
- exit and wait to be restarted
- start another connection loop with JitterUntil using max backoff timeout as the period
- abandon the attempt to send metrics

Reply:
I agree. Opted for an exit in this case to avoid subtle autoscaling bugs if the connection cannot be made at all.